cloud-minecraft/cloud-brigadier/src/main/java/cloud/commandframework/brigadier/CloudBrigadierManager.java

```diff
@@ -655,16 +656,21 @@ public void registerDefaultArgumentTypeSupplier(
             suggestionsBuilder = builder.createOffset(builder.getStart() + lastIndexOfSpaceInRemainingString + 1);
         }
 
-        for (final String suggestion : suggestions) {
-            String tooltip = argument.getName();
-            if (!(argument instanceof StaticArgument)) {
-                if (argument.isRequired()) {
-                    tooltip = '<' + tooltip + '>';
-                } else {
-                    tooltip = '[' + tooltip + ']';
+        for (final Completion completion : completions) {
+            if (completion instanceof NativeCompletion) {
```

Comment:
This is pretty clunky when working with Adventure `Component`s, since they don't implement Message at compile time. We should allow end users to inject custom handling here (and for paper/others? with platform-specific completion handling) -- to support adventure components/whatever else

Reply:
also only on velocity we can support adventure directly, paper should work without (for versions before 1.16.5), bukkit without (bukkit never had adventure), fabric without, and i can't put such utilities to extras as on every platform the adventure needs to be handled differently

Reply:
The API we expose for end users *must* include: `RichCompletion.of(String value, @Nullable Component tooltip)`. We can expose a settable function on the `CloudBrigadierManager` to perform the mapping.

Your workaround in the bukkit example is ugly af and users shouldn't have to deal with that directly -- especially if they're doing a cross-platform project, we don't want them to be forced to write their own abstraction just to produce completions with tooltips. Native platforms can set that up automatically. We could also have a fallback in `minecraft-extras` checking if the `Component` is an instanceof `Message` to have automatic handling for places where we we can do runtime class transformations (fabric, eventually forge).

Reply:
did you mean something like https://github.com/Incendo/cloud/pull/418/commits/bb4b9cd41c19b5049cce9592604bb6ed771c6106?